### PR TITLE
Add support for ubuntu 20.04

### PR DESCRIPTION
**Pull request by leonardoedgar**
- Ubuntu version: 20.04
- OpenRAVE version: 0.53
- OpenSceneGraph: 3.4
- Flexible Collision Library: 0.5.0

Tested in a docker container

### CODE REVIEW

**Comment by nicholasadr** [comment]
Cool! I notice that this commit upgrades the script compatibility to newer versions of Ubuntu and OpenRAVE (`production` branch I assume?). Out of curiosity, which changes are intended for the Ubuntu distro upgrade and which are for upgrading the OpenRAVE version? Asking in case I want to do different installation combination, e.g. Ubuntu 18 + OpenRAVE 0.53

**Comment by leonardoedgar** [comment]
> Cool! I notice that this commit upgrades the script compatibility to newer versions of Ubuntu and OpenRAVE (`production` branch I assume?). Out of curiosity, which changes are intended for the Ubuntu distro upgrade and which are for upgrading the OpenRAVE version? Asking in case I want to do different installation combination, e.g. Ubuntu 18 + OpenRAVE 0.53

Yes, it is one of the latest commit from the production branch (just committed this week). For now the script only allows the following configs:
- Ubuntu 14.04 with OpenRAVE 0.9
- Ubuntu 16.04 with OpenRAVE 0.9
- Ubuntu 18.04 with OpenRAVE 0.9
- Ubuntu 20.04 with OpenRAVE 0.53

There is only 1 OpenRAVE version for each OS, so the user will not have any options to select which OpenRAVE version to install in one OS.

File: install-dependencies.sh [hunk 1]
```diff
@@ -28,32 +30,47 @@ if [ ${UBUNTU_VER} = '14.04' ]; then
...
-g++ git ipython octave python-dev python-h5py python-numpy python-pip           \
-python-scipy python-setuptools python-wheel wget mlocate
+sudo apt-get install -y --no-install-recommends build-essential cmake doxygen \
+  g++ git octave python-dev python-setuptools wget mlocate
+if [ ${UBUNTU_VER} = '14.04' ] || [ ${UBUNTU_VER} = '16.04' ] || [ ${UBUNTU_VER} = '18.04' ]; then
+  sudo apt-get install -y --no-install-recommends ipython python-h5py python-numpy \
+    python-pip python-wheel python-scipy
+elif [ ${UBUNTU_VER} = '20.04' ]; then
+  sudo apt-get install -y --no-install-recommends python2 
+  curl https://bootstrap.pypa.io/get-pip.py --output get-pip.py
```
Comment (mstallone) [inline install-dependencies.sh:42]: On a fresh install of Ubuntu:20.04, curl is not installed so pip2 fails to install.

File: install-dependencies.sh [hunk 2]
```diff
@@ -67,5 +84,23 @@ elif [ ${UBUNTU_VER} = '18.04' ]; then
     sudo ./b2 -j `nproc` install threading=multi
 fi
 
+if [ ${UBUNTU_VER} = '20.04' ]; then
+  # Install RapidJSON
+  mkdir -p ~/git 
+  cd ~/git && git clone https://github.com/Tencent/rapidjson.git
+  cd rapidjson && mkdir build && cd build
+  cmake .. && make -j `nproc` && make install
```
Comment (mstallone) [inline install-dependencies.sh]: Installing rapidjson and pybind11 fails when running without permissions. Might suggest changing to `sudo make install` for both like osg, fcl, and openrave script files.

File: install-dependencies.sh [hunk 3]
```diff
@@ -67,5 +84,23 @@ elif [ ${UBUNTU_VER} = '18.04' ]; then
...
+
+  # Install Pybind
+  cd ~/git && git clone https://github.com/pybind/pybind11.git 
+  cd pybind11 && mkdir build && cd build 
+  git remote add woody https://github.com/woodychow/pybind11.git \
+    && git fetch woody && git checkout v2.2.4 \
+    && git cherry-pick 94824d68a037d99253b92a5b260bb04907c42355 \
+    && git cherry-pick 98c9f77e5481af4cbc7eb092e1866151461e3508 \
+    && cmake .. -DPYBIND11_TEST=OFF -DPythonLibsNew_FIND_VERSION=2 \
+    && make install
```
Comment (mstallone) [inline install-dependencies.sh]: See rapidjson comment
`sudo make install`

**Review by mstallone — CHANGES_REQUESTED** [review]
Tried this feature branch on a fresh install of Ubuntu 20.04, ran into a few QOL errors with the dependency script; otherwise, seems like it installs openrave v0.53.1 correctly.

Appreciate the work with maintaining and updating this!

**Comment by nicholasadr** [comment]
I haven't had the time to look into this but given the recent review it seems that it's good to go. Let me know if you want me (or anyone else) to merge it @leonardoedgar ?

**Comment by leonardoedgar** [comment]
> I haven't had the time to look into this but given the recent review it seems that it's good to go. Let me know if you want me (or anyone else) to merge it @leonardoedgar ?

I think we can merge this to master already.

**Comment by nicholasadr** [comment]
Thanks @leonardoedgar @mstallone, i'll merge this

**Comment by nicholasadr** [comment]
Thanks @leonardoedgar @mstallone, i'll merge this

**Pull request merged**